### PR TITLE
always run canary job

### PR DESCRIPTION
**Pull request by BenTheElder**
follow up to https://github.com/kubernetes/test-infra/pull/4263#

### CODE REVIEW

**Comment by BenTheElder** [comment]
@krzyzacy I guess not. (https://github.com/kubernetes/test-infra/pull/4263#issuecomment-326111185)

**Comment by krzyzacy** [comment]
hummmm... @spxtr should `run_after_success` kinda infer `always_run`?

File: prow/config.yaml [hunk 1]
```diff
@@ -115,6 +115,7 @@ presubmits:
     run_after_success:
     - name: pull-kubernetes-e2e-gce-canary
       agent: kubernetes
+      always_run: true
       context: pull-kubernetes-e2e-gce-canary
       rerun_command: "/test pull-kubernetes-e2e-gce-canary"
       trigger: "(?m)^/test pull-kubernetes-e2e-gce-canary,?(\\s+|$)"
```
Comment (krzyzacy) [inline prow/config.yaml]: ^^ 
/shrug

**Comment by BenTheElder** [comment]
I really need to get around to moving `/test all` into prow instead of a bazillion regexes :thinking:

**Pull request merged**